scripts/set-environment

```diff
@@ -110,6 +110,16 @@ then
   ) >> $BASH_ENV
 fi
 
+# If the Dev environment is locked, users may provide a URL-encoded
+# 'username:password' string in an environment variable to include the HTTP
+# Basic Authentication in the DEV_SITE_URL.
+if [[ -n {$DEV_SITE_BASIC_AUTH} ]]
+then
+  (
+    echo "export DEV_SITE_URL='https://$DEV_SITE_BASIC_AUTH@dev-$TERMINUS_SITE.pantheonsite.io/'"
```

Comment:
How about if you change the line line above from:
```
echo "export DEV_SITE_URL='https://dev-$TERMINUS_SITE.pantheonsite.io/'"
```
to:
```
echo "export DEV_SITE_URL='https://${DEV_SITE_BASIC_AUTH}dev-$TERMINUS_SITE.pantheonsite.io/'"
```
If "DEV_SITE_BASIC_AUTH" is not empty, then append an `@` to it as a service.

Optionally:
```
DEV_SITE_BASIC_AUTH=${DEV_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
```
That would allow folks to define either DEV_SITE_BASIC_AUTH or SITE_BASIC_AUTH. This is handy if you want to support TEST_SITE_BASIC_AUTH and LIVE_SITE_BASIC_AUTH (optional).

Reply:
Thanks @greg-1-anderson I've made the changes you've requested above and did an amend w/ forced push keep the merge clean. 

I ran with your suggestion about supporting the various environment specific variables.

```
MULTIDEV_SITE_BASIC_AUTH=${MULTIDEV_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
DEV_SITE_BASIC_AUTH=${DEV_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
TEST_SITE_BASIC_AUTH=${TEST_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
LIVE_SITE_BASIC_AUTH=${LIVE_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
```

I wasn't sure what you meant by _**append an `@` as a service**_ so I did the best I could using the `:+` shell expansion parameter one-liners like this:

```
# Append the @ sign to the Basic Authentication string.
MULTIDEV_SITE_BASIC_AUTH=${MULTIDEV_SITE_BASIC_AUTH:+"$MULTIDEV_SITE_BASIC_AUTH@"}
DEV_SITE_BASIC_AUTH=${DEV_SITE_BASIC_AUTH:+"$DEV_SITE_BASIC_AUTH@"}
TEST_SITE_BASIC_AUTH=${TEST_SITE_BASIC_AUTH:+"$TEST_SITE_BASIC_AUTH@"}
LIVE_SITE_BASIC_AUTH=${LIVE_SITE_BASIC_AUTH:+"$LIVE_SITE_BASIC_AUTH@"}
```

I also modified the documentation in the other repository and force pushed PR pantheon-systems/example-drops-8-composer#404

I tested it all locally with a simple bash script pasting the required code bits, with a small section at the top where I commented out various lines to test the different scenarios:

```
#!/bin/bash

TERMINUS_ENV='somemultidev'
TERMINUS_SITE='testing'

# Comment or uncomment the following lines to test the logic below.
# MULTIDEV_SITE_BASIC_AUTH='MULTIDEVuser:MULTIDEVpass'
DEV_SITE_BASIC_AUTH='DEVuser:DEVpass'
# TEST_SITE_BASIC_AUTH='TESTuser:TESTpass'
# LIVE_SITE_BASIC_AUTH='LIVEuser:LIVEpass'
# SITE_BASIC_AUTH='SITEuser:SITEpass'

# If any Pantheon environments are locked, users may provide a URL-encoded
# 'username:password' string in an environment variable to include the
# HTTP Basic Authentication.
MULTIDEV_SITE_BASIC_AUTH=${MULTIDEV_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
DEV_SITE_BASIC_AUTH=${DEV_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
TEST_SITE_BASIC_AUTH=${TEST_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}
LIVE_SITE_BASIC_AUTH=${LIVE_SITE_BASIC_AUTH:-$SITE_BASIC_AUTH}

# Prepare the Basic Authentication strings, appending the "@" sign if not empty.
MULTIDEV_SITE_BASIC_AUTH=${MULTIDEV_SITE_BASIC_AUTH:+"$MULTIDEV_SITE_BASIC_AUTH@"}
DEV_SITE_BASIC_AUTH=${DEV_SITE_BASIC_AUTH:+"$DEV_SITE_BASIC_AUTH@"}
TEST_SITE_BASIC_AUTH=${TEST_SITE_BASIC_AUTH:+"$TEST_SITE_BASIC_AUTH@"}
LIVE_SITE_BASIC_AUTH=${LIVE_SITE_BASIC_AUTH:+"$LIVE_SITE_BASIC_AUTH@"}

echo "export MULTIDEV_SITE_URL='https://${MULTIDEV_SITE_BASIC_AUTH}$TERMINUS_ENV-$TERMINUS_SITE.pantheonsite.io/'"
echo "export DEV_SITE_URL='https://${DEV_SITE_BASIC_AUTH}dev-$TERMINUS_SITE.pantheonsite.io/'"
echo "export TEST_SITE_URL='https://${TEST_SITE_BASIC_AUTH}test-$TERMINUS_SITE.pantheonsite.io/'"
echo "export LIVE_SITE_URL='https://${LIVE_SITE_BASIC_AUTH}live-$TERMINUS_SITE.pantheonsite.io/'"
```